### PR TITLE
Add a migration file

### PR DESCRIPTION
**Pull request by diml**
This PR adds a `MIGRATION.md` file explaining how the transition to Dune will happen with a timeline for the various steps.

### CODE REVIEW

File: MIGRATION.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
+Migrating from Jbuilder to Dune
+===============================
+
+Dune was initially called Jbuilder. Up to mid-2018, the package was
+still called `jbuilder` and it was only installing a `jbuilder`
```
Comment (hcarty) [inline MIGRATION.md]: Maybe `which only installed` instead of `and it was only installing`.

File: MIGRATION.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  `dune` files
+- `jbuild-workspace` are replaced by `dune-workspace` files
+- `jbuild-workspace<suffix>` files no longer mean anything
+
+Following are detailed explanation of the differences between the
+Jbuilder configuration files and the Dune ones.
+
+### dune-project files
+
+These are a new kind of files. With Jbuilder, projects used to be
```
Comment (hcarty) [inline MIGRATION.md]: `s/files/file/`

File: MIGRATION.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+Eventually, we are hoping that Dune will generate opam files. So users
+will only have to write a `dune-project` file.
+
+The purpose of this file is to:
+- delimit projects in larger workspaces
+- set a few project-wide parameters, such as the name, the version of
+  the Dune language in use or specification of extra features
+  (plugins) used in the project
+
+Eventually, for users who whishes to do so it should be possible to
```
Comment (hcarty) [inline MIGRATION.md]: `s/whishes/wish/`

File: MIGRATION.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+
+Following are detailed explanation of the differences between the
+Jbuilder configuration files and the Dune ones.
+
+### dune-project files
+
+These are a new kind of files. With Jbuilder, projects used to be
+identified by the presence of at least one `<package>.opam` file in a
+directory. This will still be supported until July 2019, however has
+Jbuilder evolved it became clear that we needed project files, so Dune
```
Comment (tmcgilchrist) [inline MIGRATION.md]: s/has/as

File: MIGRATION.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+users to switch their development repositories and continue their
+usual release cycle. There is no need to re-release existing packages
+just to switch to Dune immediately.
+
+The plan is as follow:
+
+### July 2018: release of Dune 1.0.0
+
+First release of the opam package `dune`. The `jbuilder` package
+becomes a transition package that depends on `dune`.
```
Comment (Chris00) [inline MIGRATION.md]: transition → transitional ?

File: MIGRATION.md [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+rest is unchanged.
+
+During this period, it makes sense for projects to do new releases
+just to switch to Dune if none of their existing releases is using
+Dune.
+
+### July 2019: support for Jbuilder is dropped
+
+`jbuilder` is now a dummy executable that always exit with an error
+message on startup. `dune` no longer reads `jbuild` or other Jbuidler
```
Comment (Chris00) [inline MIGRATION.md]: If it is the `jbuilder` _package_ is also updated to do that, one must make sure all _existing_ packages in opam are updated with an upper bound on `jbuilder` or they all become uninstallable.

Comment (diml) [reply MIGRATION.md]: The `jbuilder` package will be unchanged, but we will indeed need to add a conflict with newer versions of `dune`. I added a comment about it.

**Comment by diml** [comment]
No problem, thanks all for the reviews. I made the necessary changes

**Pull request merged**